### PR TITLE
Fix for https://github.com/Azure/azure-rest-api-specs/issues/2907

### PR DESCRIPTION
**Pull request by naveedaz**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/web/resource-manager/Microsoft.Web/stable/2016-08-01/WebApps.json [hunk 1]
```diff
@@ -12320,10 +12320,23 @@
         ],
         "responses": {
           "200": {
-            "description": "OK",
+            "description": "Successfully retrieved source control for web app.",
+            "schema": {
+              "$ref": "#/definitions/SiteSourceControl"
+            }
+          },
+          "201": {
```
Comment (jhendrixMSFT) [inline specification/web/resource-manager/Microsoft.Web/stable/2016-08-01/WebApps.json]: Does "x-ms-long-running-operation: true" need to be added to this (and the other) operation?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Need some additional feedback

**Comment by fearthecowboy** [comment]
Long running is only needed if the service isn't going to return right away -- `201 Created` signifies the creation of a new resource, but if it's instantaneous (or close enough) and the service doesn't return a `Location` header, then LRO isn't required or useful.

And, this is a `GET` operation, which does not infer creation of a resource (ie, `PUT`) , nor the beginning of a request to process something (ie, an update/operation via `PUT`/`POST`) 

I didn't see a state in the model definition for `SiteSourceControl` 

Hey @ravbhatnagar   -- Should they not be just adding some kind of provisioning state to the response object in this case (to support the scenario where the `GET` operation is called on a resource that isn't finished creating) -- or some other alternative?

**Comment by naveedaz** [comment]
@suwatch (API owner) Can you please comment on the concerns here. 
@fearthecowboy, @jhendrixMSFT FYI the API has been behaving like this since 2015. So even if the behavior is incorrect it is currently how it is implemented. We recently found out that the response codes were missing from Swagger, because customers would see these response codes and the clients would fail.

**Comment by suwatch** [comment]
This is a long running and could last for minutes/hours depending on user's deployment artifact.  As part of our GET, we do return ProvisioningState to indicate InProgress/Success/Failed.   Hope I answer the question about the api.

**Comment by dsgouda** [comment]
1. If provisioning state is being returned, it has not been modeled
2. The GET doesn't actually create a resource as indicated by `201 Created` or `202 Accepted` status codes as @fearthecowboy pointed

Also, If I understand correctly the deployment (which is `PUT`/`POST` operation) must be long running and not the `GET` operation itself

**Comment by suwatch** [comment]
I understand your points.  BTW, this long-running operation api works fine with ARM template engine since day one per https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-async-operations.     The GET may return 202 if not completed.   I think the missing part is that if the GET were called by clients other than ARM template engine, we should not be returning 201 or 202, correct?

>If the request is still running, you receive a status code 202. If the request has completed, your receive a status code 200, and the body of the response contains the properties of the storage account that has been created.

**Comment by fearthecowboy** [comment]
>FYI the API has been behaving like this since 2015

So, the current SDKs must be broken, for the GET operations.  The question is, what should we do now?

The Client Runtime doesn't have any support for GET with LRO. 

lemme go ping a couple people.

**Comment by dsgouda** [comment]
Approving this PR only as an accurate representation of the service today but this is against guidelines; will open an issue against this

**Review by dsgouda — APPROVED** [review]

**Comment by naveedaz** [comment]
@fearthecowboy Thanks for the review. Please assign the new issue to @suwatch

**Comment by dsgouda** [comment]
Opened issue to fix the service [here](https://github.com/Azure/azure-rest-api-specs/issues/2993) and issue to implement a linter rule [here](https://github.com/Azure/azure-openapi-validator/issues/162)

**Pull request merged**